src/systemTest/java/com/flightstats/hub/system/service/S3Service.java

```diff
@@ -0,0 +1,132 @@
+package com.flightstats.hub.system.service;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.AmazonS3Exception;
+import com.amazonaws.services.s3.model.GetObjectRequest;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.amazonaws.services.s3.model.S3Object;
+import com.amazonaws.services.s3.model.S3ObjectInputStream;
+
+import com.amazonaws.util.IOUtils;
+import com.flightstats.hub.model.ChannelStorage;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+import java.util.zip.ZipInputStream;
+
+import static com.flightstats.hub.system.config.PropertiesName.HELM_RELEASE_NAME;
+import static com.flightstats.hub.system.config.PropertiesName.HUB_URL_TEMPLATE;
+import static com.flightstats.hub.system.config.PropertiesName.S3_BUCKET_TEMPLATE;
+import static com.flightstats.hub.system.config.PropertiesName.TEST_DATA;
+
+@Slf4j
+public class S3Service {
+    private final String testData;
+    private final AmazonS3 s3Client;
+    private final String bucketName;
+    private final String hubBaseUrl;
+
+    @Inject
+    public S3Service(AmazonS3 s3Client,
+                     @Named(HUB_URL_TEMPLATE) String hubBaseUrl,
+                     @Named(S3_BUCKET_TEMPLATE) String bucketName,
+                     @Named(TEST_DATA) String testData,
+                     @Named(HELM_RELEASE_NAME) String releaseName) {
+        this.s3Client = s3Client;
+        this.hubBaseUrl = String.format(hubBaseUrl, releaseName);
+        this.bucketName = String.format(bucketName, releaseName);
+        this.testData = testData;
+    }
+
+    @SneakyThrows
+    private byte[] getS3BatchedItems(String path) {
+        GetObjectRequest getObjectRequest = new GetObjectRequest(bucketName, path);
+        S3Object obj = s3Client.getObject(getObjectRequest);
+        try (S3ObjectInputStream content = obj.getObjectContent()) {
+            return handleZip(content);
+        }
+    }
+
+    @SneakyThrows
+    private byte[] getS3Items(String path) {
+        GetObjectRequest getObjectRequest = new GetObjectRequest(bucketName, path);
+        S3Object obj = s3Client.getObject(getObjectRequest);
```

Comment:
Does this way ensure the stream/http connection is auto-closed? It looks kinda like the signature that takes a file path to store it /does/ auto-close, but I think you might want to use a try-with-resource starting on this line to ensure that the `Closeable` interface is fully leveraged. Maybe. I can never remember how to use this in the safest way possible.

Reply:
I think you are right, this needs to be try w resources